### PR TITLE
fix compose run exit code issue

### PR DESCRIPTION
**Pull request by carmark**
```
[lei@h8s-single hypercli]$ pkt compose run test ping -c 2 8.8.8.8
PING 8.8.8.8 (8.8.8.8): 56 data bytes
64 bytes from 8.8.8.8: seq=0 ttl=55 time=3.401 ms
64 bytes from 8.8.8.8: seq=1 ttl=55 time=2.320 ms

--- 8.8.8.8 ping statistics ---
2 packets transmitted, 2 packets received, 0% packet loss
round-trip min/avg/max = 2.320/2.860/3.401 ms
[lei@h8s-single hypercli]$ echo $?
0
[lei@h8s-single hypercli]$ pkt compose run --rm test ping -c 2 8.8.8.8
PING 8.8.8.8 (8.8.8.8): 56 data bytes
64 bytes from 8.8.8.8: seq=0 ttl=55 time=3.212 ms
64 bytes from 8.8.8.8: seq=1 ttl=55 time=2.421 ms

--- 8.8.8.8 ping statistics ---
2 packets transmitted, 2 packets received, 0% packet loss
round-trip min/avg/max = 2.421/2.816/3.212 ms
INFO[0012] Project [hypercli]: Deleting project
INFO[0012] [0/1] [test]: Deleting
INFO[0022] [0/1] [test]: Deleted
INFO[0022] Project [hypercli]: Project deleted
[lei@h8s-single hypercli]$ echo $?
0
```

```
[lei@h8s-single hypercli]$ pkt compose run test haha
exec failed: No such file or directory
[lei@h8s-single hypercli]$ echo $?
127
[lei@h8s-single hypercli]$ pkt compose run --rm test haha
exec failed: No such file or directory
INFO[0010] Project [hypercli]: Deleting project
INFO[0010] [0/1] [test]: Deleting
INFO[0012] [0/1] [test]: Deleted
INFO[0012] Project [hypercli]: Project deleted
[lei@h8s-single hypercli]$ echo $?
127
```

### CODE REVIEW

File: vendor/src/github.com/hyperhq/libcompose/docker/container.go [hunk 1]
```diff
@@ -317,20 +317,15 @@ func (c *Container) Run(ctx context.Context, imageName string, configOverride *c
...
 	if err := c.client.ContainerStart(ctx, container.ID, ""); err != nil {
-		return -1, err
+		return -1, container.ID, err
 	}
 
 	if err := <-errCh; err != nil {
 		logrus.Debugf("Error hijack: %s", err)
-		return -1, err
+		return -1, container.ID, err
 	}
```
Comment (gnawux) [inline vendor/src/github.com/hyperhq/libcompose/docker/container.go]: Why not do containerWait here? Then we do not need to change the interface, right?

Comment (carmark) [inline vendor/src/github.com/hyperhq/libcompose/docker/container.go]: we need know whether the user want to remove the container after it, they have different operations.  If we do not change the interface, we have to provide the flag of `*rm` to this function.

Comment (gnawux) [inline vendor/src/github.com/hyperhq/libcompose/docker/container.go]: What's the difference? I think wait its exist and getExitCode should work for both of the cases.

Comment (carmark) [inline vendor/src/github.com/hyperhq/libcompose/docker/container.go]: https://github.com/hyperhq/hypercli/pull/198/files/bbbf8432a5b46696adbb64f4ad8c4b7784779a11#diff-6ab99c3c122cf8ebab46b05a214b1e08R73

same as `api/client/run.go`

Comment (gnawux) [inline vendor/src/github.com/hyperhq/libcompose/docker/container.go]: Cannot get explanation from the `run` code. Please just tell why not wait stop and get exit code.

**Comment by carmark** [comment]
@gnawux updated

**Pull request merged**